server/node/src/main/java/org/kaaproject/kaa/server/operations/service/delta/DefaultDeltaService.java

```diff
@@ -185,14 +197,21 @@ private void logHashMismatch(GetDeltaRequest request, EndpointProfileDto profile
      */
     private ConfigurationCacheEntry getDelta(final String endpointId, final String userId, DeltaCacheKey deltaKey, boolean useConfigurationRawSchema) throws GetDeltaException {
         EndpointUserConfigurationDto userConfiguration = findLatestUserConfiguration(userId, deltaKey);
-
+        Optional<EndpointSpecificConfigurationDto> epsConfigOpt = Optional.empty();
```

Comment:
this method need to be improved. to many null checks while using optional. Either use Optional.isEmpty and other useful methods or use java 7 style.